### PR TITLE
DOCSUP-5169: Document the min_compress_block_size and max_compress_block_size settings

### PR DESCRIPTION
**Pull request by sevirov**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:

- Documentation for #17867

### CODE REVIEW

File: docs/ru/operations/settings/settings.md [hunk 1]
```diff
@@ -796,23 +796,27 @@ log_query_threads=1
 
 ## max_compress_block_size {#max-compress-block-size}
 
-Максимальный размер блоков не сжатых данных перед сжатием при записи в таблицу. По умолчанию - 1 048 576 (1 MiB). При уменьшении размера, незначительно уменьшается коэффициент сжатия, незначительно возрастает скорость сжатия и разжатия за счёт кэш-локальности, и уменьшается потребление оперативки. Как правило, не имеет смысла менять эту настройку.
+Максимальный размер блоков не сжатых данных перед сжатием при записи в таблицу. По умолчанию - 1 048 576 (1 MiB). При уменьшении размера, незначительно уменьшается коэффициент сжатия, незначительно возрастает скорость сжатия и разжатия за счёт кэш-локальности, и уменьшается потребление оперативки.
+
+!!! note "Внимание"
+    Эта настройка экспертного уровня, и вы не должны ее менять, если только начинаете работать с Clickhouse.
```
Comment (adevyatova) [inline docs/ru/operations/settings/settings.md:802]:
```suggestion
    Эта настройка экспертного уровня, не используйте ее, если вы только начинаете работать с Clickhouse.
```

File: docs/ru/operations/settings/settings.md [hunk 2]
```diff
@@ -796,23 +796,27 @@ log_query_threads=1
...
+Рассмотрим пример. Пусть `index_granularity`, указанная при создании таблицы - 8192.
 
-Пусть мы записываем столбец типа UInt32 (4 байта на значение). При записи 8192 строк, будет всего 32 КБ данных. Так как min_compress_block_size = 65 536, сжатый блок будет сформирован на каждые две засечки.
+Пусть мы записываем столбец типа UInt32 (4 байта на значение). При записи 8192 строк, будет всего 32 КБ данных. Так как `min_compress_block_size` = 65 536, сжатый блок будет сформирован на каждые две засечки.
 
 Пусть мы записываем столбец URL типа String (средний размер - 60 байт на значение). При записи 8192 строк, будет, в среднем, чуть меньше 500 КБ данных. Так как это больше 65 536 строк, то сжатый блок будет сформирован на каждую засечку. В этом случае, при чтении с диска данных из диапазона в одну засечку, не будет разжато лишних данных.
 
-Как правило, не имеет смысла менять эту настройку.
+!!! note "Внимание"
+    Эта настройка экспертного уровня, и вы не должны ее менять, если только начинаете работать с Clickhouse.
```
Comment (adevyatova) [inline docs/ru/operations/settings/settings.md:819]:
```suggestion
    Эта настройка экспертного уровня, не используйте ее, если вы только начинаете работать с Clickhouse.
```

File: docs/ru/operations/settings/settings.md [hunk 3]
```diff
@@ -796,23 +796,27 @@ log_query_threads=1
...
-Рассмотрим пример. Пусть index_granularity, указанная при создании таблицы - 8192.
+Рассмотрим пример. Пусть `index_granularity`, указанная при создании таблицы - 8192.
 
-Пусть мы записываем столбец типа UInt32 (4 байта на значение). При записи 8192 строк, будет всего 32 КБ данных. Так как min_compress_block_size = 65 536, сжатый блок будет сформирован на каждые две засечки.
+Пусть мы записываем столбец типа UInt32 (4 байта на значение). При записи 8192 строк, будет всего 32 КБ данных. Так как `min_compress_block_size` = 65 536, сжатый блок будет сформирован на каждые две засечки.
 
 Пусть мы записываем столбец URL типа String (средний размер - 60 байт на значение). При записи 8192 строк, будет, в среднем, чуть меньше 500 КБ данных. Так как это больше 65 536 строк, то сжатый блок будет сформирован на каждую засечку. В этом случае, при чтении с диска данных из диапазона в одну засечку, не будет разжато лишних данных.
 
-Как правило, не имеет смысла менять эту настройку.
+!!! note "Внимание"
```
Comment (adevyatova) [inline docs/ru/operations/settings/settings.md:818]:
```suggestion
!!! note "Предупреждение"
```

File: docs/ru/operations/settings/settings.md [hunk 4]
```diff
@@ -796,23 +796,27 @@ log_query_threads=1
 
 ## max_compress_block_size {#max-compress-block-size}
 
-Максимальный размер блоков не сжатых данных перед сжатием при записи в таблицу. По умолчанию - 1 048 576 (1 MiB). При уменьшении размера, незначительно уменьшается коэффициент сжатия, незначительно возрастает скорость сжатия и разжатия за счёт кэш-локальности, и уменьшается потребление оперативки. Как правило, не имеет смысла менять эту настройку.
+Максимальный размер блоков не сжатых данных перед сжатием при записи в таблицу. По умолчанию - 1 048 576 (1 MiB). При уменьшении размера, незначительно уменьшается коэффициент сжатия, незначительно возрастает скорость сжатия и разжатия за счёт кэш-локальности, и уменьшается потребление оперативки.
+
+!!! note "Внимание"
```
Comment (adevyatova) [inline docs/ru/operations/settings/settings.md:801]:
```suggestion
!!! note "Предупреждение"
```

File: docs/ru/engines/table-engines/mergetree-family/mergetree.md [hunk 5]
```diff
@@ -77,17 +77,19 @@ ORDER BY expr
...
+    -   `write_final_mark` — включает или отключает запись последней засечки индекса в конце куска данных, указывающей за последний байт. По умолчанию — 1. Не отключайте её.
+    -   `merge_max_block_size` — максимальное количество строк в блоке для операций слияния. Значение по умолчанию: 8192.
+    -   `storage_policy` — политика хранения данных. Смотрите [Хранение данных таблицы на нескольких блочных устройствах](#table_engine-mergetree-multiple-volumes).
     -   `min_bytes_for_wide_part`, `min_rows_for_wide_part` — минимальное количество байт/строк в куске данных для хранения в формате `Wide`. Можно задать одну или обе настройки или не задавать ни одной. Подробнее см. в разделе [Хранение данных](#mergetree-data-storage).
+	-   `max_compress_block_size` — максимальный размер блоков несжатых данных перед сжатием для записи в таблицу. Вы также можете указать эту настройку в глобальных настройках (смотрите настройку [max_compress_block_size](../../../operations/settings/settings.md#max-compress-block-size)). Настройка, которая задается при создании таблицы, имеет более высокий приоритет, чем глобальная настройка.
```
Comment (adevyatova) [inline docs/ru/engines/table-engines/mergetree-family/mergetree.md:91]:
```suggestion
	-   `max_compress_block_size` — максимальный размер блоков несжатых данных перед сжатием для записи в таблицу. Вы также можете задать этот параметр в глобальных настройках (смотрите [max_compress_block_size](../../../operations/settings/settings.md#max-compress-block-size)). Настройка, которая задается при создании таблицы, имеет более высокий приоритет, чем глобальная.
```

File: docs/ru/engines/table-engines/mergetree-family/mergetree.md [hunk 6]
```diff
@@ -77,17 +77,19 @@ ORDER BY expr
...
+	-   `min_compress_block_size` — минимальный размер блоков несжатых данных, необходимых для сжатия при записи следующей метки. Вы также можете указать эту настройку в глобальных настройках (смотрите настройку [min_compress_block_size](../../../operations/settings/settings.md#min-compress-block-size)). Настройка, которая задается при создании таблицы, имеет более высокий приоритет, чем глобальная настройка.
```
Comment (adevyatova) [inline docs/ru/engines/table-engines/mergetree-family/mergetree.md:92]:
```suggestion
	-   `min_compress_block_size` — минимальный размер блоков несжатых данных, необходимых для сжатия при записи следующей метки. Вы также можете задать этот параметр в глобальных настройках (смотрите [min_compress_block_size](../../../operations/settings/settings.md#min-compress-block-size)). Настройка, которая задается при создании таблицы, имеет более высокий приоритет, чем глобальная.
```

**Comment by sevirov** [comment]
@Akazz Please, review this PR.

File: docs/ru/operations/settings/settings.md [hunk 7]
```diff
@@ -796,23 +796,27 @@ log_query_threads=1
 
 ## max_compress_block_size {#max-compress-block-size}
 
-Максимальный размер блоков не сжатых данных перед сжатием при записи в таблицу. По умолчанию - 1 048 576 (1 MiB). При уменьшении размера, незначительно уменьшается коэффициент сжатия, незначительно возрастает скорость сжатия и разжатия за счёт кэш-локальности, и уменьшается потребление оперативки. Как правило, не имеет смысла менять эту настройку.
+Максимальный размер блоков не сжатых данных перед сжатием при записи в таблицу. По умолчанию - 1 048 576 (1 MiB). При уменьшении размера, незначительно уменьшается коэффициент сжатия, незначительно возрастает скорость сжатия и разжатия за счёт кэш-локальности, и уменьшается потребление оперативки.
```
Comment (Akazz) [inline docs/ru/operations/settings/settings.md:799]:
```suggestion
Максимальный размер блоков несжатых данных перед сжатием при записи в таблицу. По умолчанию - 1 048 576 (1 MiB). При уменьшении размера, незначительно уменьшается коэффициент сжатия, незначительно возрастает скорость сжатия и разжатия за счёт кэш-локальности, и уменьшается потребление оперативной памяти.
```

File: docs/ru/engines/table-engines/mergetree-family/mergetree.md [hunk 8]
```diff
@@ -77,17 +77,19 @@ ORDER BY expr
...
     -   `min_bytes_for_wide_part`, `min_rows_for_wide_part` — минимальное количество байт/строк в куске данных для хранения в формате `Wide`. Можно задать одну или обе настройки или не задавать ни одной. Подробнее см. в разделе [Хранение данных](#mergetree-data-storage).
+	-   `max_compress_block_size` — максимальный размер блоков несжатых данных перед сжатием для записи в таблицу. Вы также можете задать этот параметр в глобальных настройках (смотрите [max_compress_block_size](../../../operations/settings/settings.md#max-compress-block-size)). Настройка, которая задается при создании таблицы, имеет более высокий приоритет, чем глобальная.
+	-   `min_compress_block_size` — минимальный размер блоков несжатых данных, необходимых для сжатия при записи следующей метки. Вы также можете задать этот параметр в глобальных настройках (смотрите [min_compress_block_size](../../../operations/settings/settings.md#min-compress-block-size)). Настройка, которая задается при создании таблицы, имеет более высокий приоритет, чем глобальная.
```
Comment (Akazz) [inline docs/ru/engines/table-engines/mergetree-family/mergetree.md:92]:
```suggestion
	-   `min_compress_block_size` — минимальный размер блоков несжатых данных, необходимых для сжатия при записи следующей засечки. Вы также можете задать этот параметр в глобальных настройках (смотрите [min_compress_block_size](../../../operations/settings/settings.md#min-compress-block-size)). Настройка, которая задается при создании таблицы, имеет более высокий приоритет, чем глобальная.
```

File: docs/en/engines/table-engines/mergetree-family/mergetree.md [hunk 9]
```diff
@@ -98,7 +98,9 @@ For a description of parameters, see the [CREATE query description](../../../sql
     -   `merge_max_block_size` — Maximum number of rows in block for merge operations. Default value: 8192.
     -   `storage_policy` — Storage policy. See [Using Multiple Block Devices for Data Storage](#table_engine-mergetree-multiple-volumes).
     -   `min_bytes_for_wide_part`, `min_rows_for_wide_part` — Minimum number of bytes/rows in a data part that can be stored in `Wide` format. You can set one, both or none of these settings. See [Data Storage](#mergetree-data-storage).
-    -   `max_parts_in_total` — Maximum number of parts in all partitions. 
+    -   `max_parts_in_total` — Maximum number of parts in all partitions.
+	-   `max_compress_block_size` — Maximum size of blocks of uncompressed data before compressing for writing to a table. You can also specify this setting in the global settings (see [max_compress_block_size](../../../operations/settings/settings.md#max-compress-block-size) setting). But the setting that is set when creating the table has a higher priority than the global setting.
```
Comment (Akazz) [inline docs/en/engines/table-engines/mergetree-family/mergetree.md:102]:
```suggestion
	-   `max_compress_block_size` — Maximum size of blocks of uncompressed data before compressing for writing to a table. You can also specify this setting in the global settings (see [max_compress_block_size](../../../operations/settings/settings.md#max-compress-block-size) setting). The value specified upon table creation overrides the global value for this setting.
```

File: docs/en/engines/table-engines/mergetree-family/mergetree.md [hunk 10]
```diff
@@ -98,7 +98,9 @@ For a description of parameters, see the [CREATE query description](../../../sql
...
+	-   `min_compress_block_size` — Minimum size of blocks of uncompressed data required for compression when writing the next mark. You can also specify this setting in the global settings (see [min_compress_block_size](../../../operations/settings/settings.md#min-compress-block-size) setting). But the setting that is set when creating the table has a higher priority than the global setting.
```
Comment (Akazz) [inline docs/en/engines/table-engines/mergetree-family/mergetree.md:103]:
```suggestion
	-   `min_compress_block_size` — Minimum size of blocks of uncompressed data required for compression when writing the next mark. You can also specify this setting in the global settings (see [min_compress_block_size](../../../operations/settings/settings.md#min-compress-block-size) setting). The value specified upon table creation overrides the global value for this setting..
```

**Comment by sevirov** [comment]
@Akazz Please, review this PR.

**Pull request merged**